pkg/insights/insightsclient/insightsclient.go

```diff
@@ -130,6 +129,19 @@ func userAgent(releaseVersionEnv string, v apimachineryversion.Info, cv *configv
 	return fmt.Sprintf("insights-operator/%s cluster/%s", gitVersion, cv.Spec.ClusterID)
 }
 
+func (c *Client) getClusterVersion() (*configv1.ClusterVersion, error) {
+	if c.clusterVersion != nil {
+		return c.clusterVersion, nil
+	}
+	ctx := context.Background()
+	cv, err := clusterconfig.GetClusterVersion(ctx, c.gatherKubeConfig)
```

Comment:
I am not sure I like this dependency to `clusterconfig`. Please take a look at this https://github.com/RedHatInsights/insights-ingress-http-client/blob/main/insights/insightsclient/insightsclient.go#L149. It would probably mean to pass the clusterversion in `operator.go`. Can you please try it?

Reply:
yeah, but if I "just" pass in the version to the client it still has to come from somewhere (something needs to gather it), so that is why it uses the clusterconfig.GetClusterVersion that is just the base logic to get the cluster version which as independent as it gets,
if we plan to move the insightsclient into separate project then a copy of that gather logic will need to go with it

Reply:
Yes good point. It would probably make more sense to move the cluster version gathering to a potential new client lib I created https://github.com/RedHatInsights/insights-ingress-http-client/issues/9